pages/user.js

```diff
@@ -34,9 +71,125 @@ class User extends Component {
         return { isLoggedIn, userData };
     }
 
+    componentDidMount() {
+        this.handleSocket();
+    }
+
+    onCloseNotification() {
+        const { notificationKey: key } = this.state;
+        this.acceptChallengeRequest();
+        notification.close(key);
+    }
+
+    getUserList() {
+        this.socket.emit(getUser, {});
+    }
+
+    rejectChallengeRequest() {
+        this.socket.emit(rejectChallenge, {});
+    }
+
+    acceptChallengeRequest() {
+        this.socket.emit(acceptChallenge, {});
+    }
+
+    // call this when user want to end current challenge
+    leaveChat() {
+        const { connected, room } = this.state;
+        if (connected) {
+            this.socket.emit(leaveRoom);
+            this.socket.leave(room);
+            this.setState({ room: '' });
+        }
+    }
+
+    handleSocket() {
+        const { connect } = CLIENT_SYSTEM_EVENTS;
+        const { userData: user, isLoggedIn } = this.props;
+        const { room } = this.state;
+        const { _id, name, picture } = user;
+        this.socket = io();
+
+        this.socket.on(connect, () => {
+            this.setState({ connected: true });
+
+            if (isLoggedIn) {
+                this.socket.emit(login, { _id, name, picture });
+            }
+        });
+
+        this.socket.on(users, data => {
+            this.setState({ onlineUsers: data.users });
+            const { onlineUsers } = this.state;
+            console.log(`users called ${onlineUsers.length}`);
```

Comment:
remove all `console.log`

Reply:
This won't be added to our code. `pages/user.js` is just a test file tosin works with. I will need it as a guide when I'm properly implementing the code there.

Reply:
tl;dr I need the console.logs